src/briefcase/platforms/windows/__init__.py

```diff
@@ -124,14 +127,145 @@ def default_packaging_format(self):
     def verify_tools(self):
         super().verify_tools()
         WiX.verify(self.tools)
+        if getattr(self, "_windows_sdk_needed", False):
+            WindowsSDK.verify(self.tools)
 
-    def package_app(self, app: BaseConfig, **kwargs):
+    def add_options(self, parser):
+        super().add_options(parser)
+        parser.add_argument(
+            "--file-digest",
+            help="File digest algorithm to use for creating file signatures; defaults to sha256.",
+            default="sha256",
+            required=False,
+        )
+        parser.add_argument(
+            "--cert-store-location",
+            help="Location of stores containing the certificate; defaults to Current User.",
+            choices=["Current User", "Local Machine"],
+            default="Current User",
+            required=False,
+        )
+        parser.add_argument(
+            "--cert-store",
+            help="Store containing the certificate; defaults to the Personal store.",
+            default="My",
+            required=False,
+        )
+        parser.add_argument(
+            "--timestamp-url",
+            help=(
+                "URL for the Timestamp Authority server to timestamp "
+                "the code signing; defaults to timestamp.digicert.com."
+            ),
+            default="http://timestamp.digicert.com",
+            required=False,
+        )
+        parser.add_argument(
+            "--timestamp-digest",
+            help="Digest algorithm to request for the timestamp server signature; defaults to sha256.",
+            default="sha256",
+            required=False,
+        )
+
+    def parse_options(self, extra):
+        """Require the Windows SDK tool if an `identity` is specified."""
+        options = super().parse_options(extra=extra)
+        self._windows_sdk_needed = options["identity"] is not None
```

Comment:
I'm not overly concerned about this - it makes as much sense as anything to me. The only thing I'd tweak is (a) adding a `clone_options` implementation so that the `_windows_sdk_needed` flag is carried over in a copy, and (b) needing a `getattr()` check rather than a hard `self._windows_sdk_needed`. I suspect that might be related to the missing `clone_options`?

Reply:
I held back cloning the `_windows_sdk_needed` property to other commands because the reason the `PackageCommand` needs the Windows SDK is specific to its use-case; other commands wouldn't inherently need the Windows SDK for the same reason.....or at least it would likely be misleading...especially if other use cases for the Windows SDK are added to other commands.

I used `getattr()` elsewhere to reference `_windows_sdk_needed` because the tests invoke methods directly and don't ensure `parse_options()` is called first. Although, I could just set `_windows_sdk_needed` manually in the test to avoid an error.

Reply:
I guess I don't see why it needs to have impact outside the Package command. If the `verify_tools()` implementation is command specific, and the `parse_option` implementation is command specific... where will it be visible on other commands?

Reply:
It doesn't _need_ to have an impact outside the Package command. I was just saying I don't think I see value in cloning `_windows_sdk_needed` in a `clone_options()` implementation because the need for the Windows SDK outside of packaging a Windows app in other commands would likely be dependent on other factors.

This is already notable in the Build command for Windows apps because it is currently using the Windows SDK simply if it is available at all.....so, use in the Build command is independent of whether a user is attempting to code sign or not.

Or did I miss what you were saying about "a `clone_options` implementation" maybe...?

Reply:
You're right that clone_options probably doesn't have any impact right now; it would only matter if there was a command that invoked `package` as a pre-requisite. It's needed for `use_docker` because if you invoke run, and that implies a build, the build needs to inherit the `use_docker`. There's nothing that will imply `package`... but if anything *did* - say, for example, `publish`, then defensively including `clone_options` now would avoid the problem.

Modifying the test to ensure `_windows_sdk_needed` is always available would also be my preferred option, for similar reasons - it's defensive programming against inadvertently forgetting the significance of having a verified SDK. I'm not sure I have any particular failure mode in mind here - but I have a very poor imagination :-)

Reply:
I think I'm still confused though...so, apologies for going around again.

If I add a `clone_options()` to `WindowsPackageCommand` for `_windows_sdk_needed`, that will ensure `_windows_sdk_needed` is propagated down to pre-requisite commands; however, that would only happen for a literal `briefcase package windows` command...and we currently know the reason the Package command needs the SDK does not (and seems unlikely to ever) overlap with reasons a different Command would need it.

If `briefcase publish` ever invokes the Package command, it would need to handle this separately anyway. (And a lot more changes on top of that to accept all the signing CLI options so it could pass those to the Package command as well.)

OTOH, if you're saying, like the design for `use_docker` for Linux, I should bake in `_windows_sdk_needed` much deeper in to the Windows Command class hierarchy so _any_ command would propagate this flag, I guess that's possible....it just feels a bit like a premature optimization.

Ultimately, I guess, the comparison to `use_docker` seems inaccurate; there's a clear understanding that setting clearly applies to _all_ Commands for the app. Here, with `_windows_sdk_needed`, we have a very specific use-case for the Package command....and already the Build command has different criteria to require the SDK.

Anyway...lotta words spent on this; please let me know if I'm still not understanding your perspective.

P.S. got rid of the `getattr()`....that's the better approach.

Reply:
> I think I'm still confused though...so, apologies for going around again.
> 
> If I add a `clone_options()` to `WindowsPackageCommand` for `_windows_sdk_needed`, that will ensure `_windows_sdk_needed` is propagated down to pre-requisite commands; however, that would only happen for a literal `briefcase package windows` command...and we currently know the reason the Package command needs the SDK does not (and seems unlikely to ever) overlap with reasons a different Command would need it.

True - but "Does the system have the Windows SDK" is a fairly abstract, Windows-specific idea. It doesn't matter why that flag has been turned on - if it's on, you need the SDK to be present. 

> If `briefcase publish` ever invokes the Package command, it would need to handle this separately anyway. (And a lot more changes on top of that to accept all the signing CLI options so it could pass those to the Package command as well.)

Oh, sure - no argument there. I'm not sure chaining publication would ever make sense because of the complexity of packaging arguments.
 
> OTOH, if you're saying, like the design for `use_docker` for Linux, I should bake in `_windows_sdk_needed` much deeper in to the Windows Command class hierarchy so _any_ command would propagate this flag, I guess that's possible....it just feels a bit like a premature optimization.

That wasn't what I was suggesting, but it doesn't sound like the *worst* idea. However, I also agree it sounds like premature optimisation.
 
> Ultimately, I guess, the comparison to `use_docker` seems inaccurate; there's a clear understanding that setting clearly applies to _all_ Commands for the app. Here, with `_windows_sdk_needed`, we have a very specific use-case for the Package command....and already the Build command has different criteria to require the SDK.

I see what you're saying - I think the area of non-alignment is whether the `windows_sdk_needed` is a flag indicating something we're verifying, or something indicating the reason why we're verifying it. 

In the case of `use_docker`, the two are the same - you need to verify docker because you have to use docker. There's no ambiguity there.

In this case, we're verifying the Windows SDK because we want to do code signing - but you can/could use the Windows SDK for reasons other than code signing. My argument is that `windows_sdk_needed` is flagging a system requirement, which is true regardless of what you're doing with the SDK, so it doesn't hurt to propagate it. AFAICT, you're looking at `windows_sdk_needed` as more of "command does code signing" indicator, which doesn't mean much outside of the Package command. 

From my perspective, `clone_options` is a 4 line method that isn't *untrue* in the global context as a pure statement of system requirements, and might save us some headaches in future trying to work out why an SDK requirement isn't propagating in certain command invocations.

> Anyway...lotta words spent on this

No argument here :-) Honestly, I think the number of words expended here has massively outweighed the debugging effort that would be associated with fixing the problem if it were to occur. - so I'm happy to call adding `clone_options` an optional suggestion (... but I reserve the right to say "I told you so" if it becomes a problem in future :-P ).

Reply:
Ahh, ok; I think I see now. Thanks.

So, philosophically, I think we're on the same page. If the Package command already decided it needs the SDK, just clone it to other Commands. I agree with this.

Although, considering the implementation...
1) The only real world benefit I can think of is allowing other Commands to short circuit their own evaluation logic to determine if they need the SDK.
2) As a consequence of 1, `windows_sdk_needed` is only half reliable since it being set to `False` doesn't necessarily mean anything...if another command conditionally needs the SDK, it needs to perform its own evaluation....and even if it doesn't need it conditionally, it probably needs to verify the SDK regardless of `windows_sdk_needed`.
3) The Package command will verify the SDK before even creating the other Commands.
4) The other Commands need `windows_sdk_needed` initialized in the absence of the Package Command...the simplest approach might be setting it in a new `parse_options()` in `WindowsMixin`...

So, yeah...the `windows_sdk_needed` flag really just means "the package command needs to do code signing"...for now anyway.

All right...I think I'll leave this out for now :)...I'll take the told ya so later if need be haha